File: trunk/EngineUAB/Run/Videogame/Data/Scripts/Game/Player/player_animation_defense_state.lua
class 'CPlayerAnimationDefenseState' (CState)
	function CPlayerAnimationDefenseState:__init(name) 
		CState.__init(self, name)
	end

	function CPlayerAnimationDefenseState:OnEnter(_CCharacter)
		if not ( _CCharacter == nil ) then
			num = _CCharacter:get_animation_id("defense")
			_CCharacter:get_animation_model():blend_cycle( num, 0.2 )
		end
	end
	
	function CPlayerAnimationDefenseState:Execute(_CCharacter, _elapsed_time)
	end
	
	function CPlayerAnimationDefenseState:OnExit(_CCharacter)
		if not ( _CCharacter == nil ) then
			num = _CCharacter:get_animation_id("defense")
			_CCharacter:get_animation_model():clear_cycle( num, 0.2 )
		end
	end
	
	function CPlayerAnimationDefenseState:OnMessage(_CCharacter, _Msg)
		return false
	end
	
	function CPlayerAnimationDefenseState:__Finalize()
	end

File: trunk/EngineUAB/Code/Videogame/Characters/Player/AnimationStates/PlayerAnimationDefenseState.cpp
#include "PlayerAnimationDefenseState.h"
#include "StatesMachine\Telegram.h"
#include "RenderableObjects\AnimatedModel\AnimatedInstanceModel.h"

#if defined _DEBUG
#include "Memory\MemLeaks.h"
#endif

CPlayerAnimationDefenseState::CPlayerAnimationDefenseState( CCharacter* _pCharacter,const std::string &_Name )
	: CState(_pCharacter, _Name)
{

}

CPlayerAnimationDefenseState::~CPlayerAnimationDefenseState()
{

}

void CPlayerAnimationDefenseState::OnEnter( CCharacter* _pCharacter )
{
	int l_iAnimID = _pCharacter->GetAnimationID("defense");
	_pCharacter->GetAnimatedModel()->BlendCycle(l_iAnimID, 0.2f);
}

void CPlayerAnimationDefenseState::Execute( CCharacter* _pCharacter, float _fElapsedTime )
{

}

void CPlayerAnimationDefenseState::OnExit( CCharacter* _pCharacter )
{
	int l_iAnimID = _pCharacter->GetAnimationID("defense");
	_pCharacter->GetAnimatedModel()->ClearCycle(l_iAnimID, 0.2f);
}

bool CPlayerAnimationDefenseState::OnMessage( CCharacter* _pCharacter, const STelegram& _Message )
{
	return false;
}